storagenode/storagenodedb/usedserials.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storagenodedb
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/storj"
+)
+
+type usedSerials struct {
+	*infodb
+}
+
+// UsedSerials returns certificate database.
+func (db *infodb) UsedSerials() usedSerials { return usedSerials{db} }
+
+// Add adds a serial to the database.
+func (db *usedSerials) Add(ctx context.Context, satelliteID storj.NodeID, serialNumber []byte, expiration time.Time) error {
+	defer db.locked()()
+
+	_, err := db.db.Exec(`
+		INSERT INTO 
+			used_serial(satellite_id, serial_number, expiration) 
+		VALUES(?, ?, ?)`, satelliteID, serialNumber, expiration)
+
+	return ErrInfo.Wrap(err)
+}
+
+// DeleteExpired deletes expired serial numbers
+func (db *usedSerials) DeleteExpired(ctx context.Context, now time.Time) error {
+	defer db.locked()()
+
+	_, err := db.db.Exec(`DELETE FROM used_serial WHERE expiration < ?`, now)
+
+	return ErrInfo.Wrap(err)
+}
+
+// SerialNumberFn is callback from IterateAll
+type SerialNumberFn func(satelliteID storj.NodeID, serialNumber []byte, expiration time.Time)
+
+// IterateAll iterates all serials.
+// Note, this will lock the database and should only be used during startup.
+func (db *usedSerials) IterateAll(ctx context.Context, fn SerialNumberFn) (err error) {
```

Comment:
What does this function provide?
Ensuring that the database is healthy?

Reply:
When we restart the storagenode we need to reload all the serial numbers into memory.

Reply:
Oh yeah, good idea.